README.md
# DayNightTimePicker

<!-- ALL-CONTRIBUTORS-BADGE:START - Do not remove or modify this section -->
[![All Contributors](https://img.shields.io/badge/all_contributors-24-orange.svg?style=flat)](#contributors)
<!-- ALL-CONTRIBUTORS-BADGE:END -->

A day night time picker for Flutter with **Zero Dependencies**.

> ### Default style:
>
> <img src="https://raw.githubusercontent.com/subhamayd2/day_night_time_picker/master/example.gif" style="width: 100%; max-width: 400px; height: auto" />

<br />
<br />

> ### IOS style:
>
> <img src="https://raw.githubusercontent.com/subhamayd2/day_night_time_picker/master/example_ios_style.png" style="width: 100%; max-width: 400px; height: auto" />

<br />
<br />

> ### [View it on pub.dev](https://pub.dev/packages/day_night_time_picker)

---

## Installation

Add to pubspec.yaml.

```yaml
dependencies:
  day_night_time_picker:
```

---

## Usage

To use plugin, just import package

```dart
import 'package:day_night_time_picker/day_night_time_picker.dart';
```

---

## Example

```dart
TextButton(
    onPressed: () {
        Navigator.of(context).push(
            showPicker(
                context: context,
                value: _time,
                sunrise: TimeOfDay(hour: 6, minute: 0), // optional
                sunset: TimeOfDay(hour: 18, minute: 0), // optional
                duskSpanInMinutes: 120, // optional
                onChange: onTimeChanged,
            ),
        );
    },
    child: Text(
        "Open time picker",
        style: TextStyle(color: Colors.white),
    ),
),
```

---

## Props

| Name                            | Description                                                                                                                        |                         Default                          |
|:--------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------| :------------------------------------------------------: |
| **value**                       | **`Required`** Display value. It takes in [Time].                                                                                  |                                                          |
| **onChange**                    | **`Required`** Return the new time the user picked as [Time].                                                                      |                                                          |
| **isInlinePicker**              | Whether to render an inline widget.                                                                                                |                         `false`                          |
| **onChangeDateTime**            | _`Optional`_ Return the new time the user picked as [DateTime].                                                                    |                                                          |
| **onCancel**                    | _`Optional`_ Custom callback for the Cancel button. Note: if provided, it will override the default behavior of the Cancel button. |                                                          |
| **is24HrFormat**                | Show the time in TimePicker in 24 hour format.                                                                                     |                         `false`                          |
| **accentColor**                 | Accent color of the TimePicker.                                                                                                    |             `Theme.of(context).accentColor`              |
| **unselectedColor**             | Color applied unselected options (am/pm, hour/minute).                                                                             |                      `Colors.grey`                       |
| **cancelText**                  | Text displayed for the Cancel button.                                                                                              |                         `cancel`                         |
| **okText**                      | Text displayed for the Ok button.                                                                                                  |                           `ok`                           |
| **sunAsset**                    | Image asset used for the Sun.                                                                                                      |                      Asset provided                      |
| **moonAsset**                   | Image asset used for the Moon.                                                                                                     |                      Asset provided                      |
| **blurredBackground**           | Whether to blur the background of the [Modal].                                                                                     |                         `false`                          |
| **barrierColor**                | Color of the background of the [Modal].                                                                                            |                     `Colors.black45`                     |
| **borderRadius**                | Border radius of the [Container] in [double].                                                                                      |                          `10.0`                          |
| **elevation**                   | Elevation of the [Modal] in [double].                                                                                              |                          `12.0`                          |
| **dialogInsetPadding**          | Inset padding of the [Modal] in EdgeInsets.                                                                                        | `EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0)` |
| **barrierDismissible**          | Whether clicking outside should dismiss the [Modal].                                                                               |                          `true`                          |
| **showCancelButton**            | Whether to show the cancel button or not                                                                                           |                          `true`                          |
| **iosStylePicker**              | Whether to display a IOS style picker (Not exactly the same).                                                                      |                         `false`                          |
| **hourLabel**                   | The label to be displayed for `hour` picker. Only for _iosStylePicker_.                                                            |                        `'hours'`                         |
| **minuteLabel**                 | The label to be displayed for `minute` picker. Only for _iosStylePicker_.                                                          |                       `'minutes'`                        |
| **minuteInterval**              | Steps interval while changing `minute`. Accepts `TimePickerInterval` enum.                                                         |                 `TimePickerInterval.ONE`                 |
| **secondInterval**              | Steps interval while changing `minute`. Accepts `TimePickerInterval` enum.                                                         |                 `TimePickerInterval.ONE`                 |
| **disableMinute**               | Disables the minute picker.                                                                                                        |                         `false`                          |
| **disableHour**                 | Disables the hour picker.                                                                                                          |                         `false`                          |
| **minHour**                     | Selectable minimum hour.                                                                                                           |            Defaults to `1`[12hr] or `0`[24hr]            |
| **maxHour**                     | Selectable maximum hour.                                                                                                           |           Defaults to `12`[12hr] or `23`[24hr]           |
| **minMinute**                   | Selectable minimum minute.                                                                                                         |                           `0`                            |
| **minSecond**                   | Selectable minimum second.                                                                                                         |                           `0`                            |
| **maxMinute**                   | Selectable maximum minute.                                                                                                         |                           `59`                           |
| **maxSecond**                   | Selectable maximum second.                                                                                                         |                           `59`                           |
| **displayHeader**               | Whether to display the sun moon animation.                                                                                         |                          `true`                          |
| **isOnValueChangeMode**         | Whether to hide okText, cancelText and return value on every onValueChange. **_Only for Inline widget_**                           |                         `false`                          |
| **focusMinutePicker**           | Whether or not the minute picker is auto focus/selected.                                                                           |                         `false`                          |
| **themeData**                   | ThemeData to use for the widget.                                                                                                   |                   `Theme.of(context)`                    |
| **okStyle**                     | Ok button's text style.                                                                                                            |      `const TextStyle(fontWeight: FontWeight.bold)`      |
| **cancelStyle**                 | Cancel button's text style.                                                                                                        |      `const TextStyle(fontWeight: FontWeight.bold)`      |
| **buttonStyle**                 | Button's style. If `cancelButtonStyle` not provided, it applies to ok and cancel buttons.                                          |        `Theme.of(context).colorScheme.secondary`         |
| **cancelButtonStyle**           | Cancel button style.                                                                                                               |        `Theme.of(context).colorScheme.secondary`         |
| **buttonsSpacing**              | Spacing between cancel and ok buttons.                                                                                             |                           `0`                            |
| **wheelHeight**                 | Only for `createInlinePicker` Height of the Wheel                                                                                  |                          `240`                           |
| **hideButtons**                 | Whether to hide the buttons (ok and cancel).                                                                                       |                         `false`                          |
| **disableAutoFocusToNextInput** | Whether to disable the auto focus to the next input after current input is selected.                                               |                         `false`                          |
| **width**                       | Fixed width of the Picker container.                                                                                               |                `300` (`350` for iosStyle)                |
| **height**                      | Fixed height of the Picker container.                                                                                              |                          `400`                           |
| **sunrise**                     | Sunrise time.                                                                                                                      |                          `400`                           |
| **sunset**                      | Sunset time.                                                                                                                       |                          `400`                           |
| **duskSpanInMinutes**           | Dusk span in minutes.                                                                                                              |                          `400`                           |
| **settings**                    | _`Optional`_ Data that might be useful in constructing a [Route].                                                                  |                                                          |

---

## Contributors

Thanks goes to these wonderful people:

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<table>
  <tbody>
    <tr>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/subhamayd2"><img src="https://avatars.githubusercontent.com/u/23093995?v=4?s=100" width="100px;" alt="Subhamay Dutta"/><br /><sub><b>Subhamay Dutta</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=subhamayd2" title="Code">💻</a> <a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=subhamayd2" title="Documentation">📖</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/impure"><img src="https://avatars.githubusercontent.com/u/4359114?v=4?s=100" width="100px;" alt="Andrew Zuo"/><br /><sub><b>Andrew Zuo</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=impure" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://www.linkedin.com/in/oude-mohammad/"><img src="https://avatars.githubusercontent.com/u/6555426?v=4?s=100" width="100px;" alt="Mohammad Odeh"/><br /><sub><b>Mohammad Odeh</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=MOOUDE" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/hashem78"><img src="https://avatars.githubusercontent.com/u/4525797?v=4?s=100" width="100px;" alt="Hashem Alayan"/><br /><sub><b>Hashem Alayan</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=hashem78" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/gohdong"><img src="https://avatars.githubusercontent.com/u/22044475?v=4?s=100" width="100px;" alt="gohdong"/><br /><sub><b>gohdong</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=gohdong" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://achim.io/"><img src="https://avatars.githubusercontent.com/u/43643339?v=4?s=100" width="100px;" alt="nohli"/><br /><sub><b>nohli</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=nohli" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/sander102907"><img src="https://avatars.githubusercontent.com/u/22891388?v=4?s=100" width="100px;" alt="sander102907"/><br /><sub><b>sander102907</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=sander102907" title="Code">💻</a></td>
    </tr>
    <tr>
      <td align="center" valign="top" width="14.28%"><a href="https://asoteam.ir"><img src="https://avatars.githubusercontent.com/u/22625638?v=4?s=100" width="100px;" alt="Sobhan Moradi"/><br /><sub><b>Sobhan Moradi</b></sub></a><br /><a href="#design-sobimor" title="Design">🎨</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/iaskari"><img src="https://avatars.githubusercontent.com/u/3792357?v=4?s=100" width="100px;" alt="Omar Dahhane"/><br /><sub><b>Omar Dahhane</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=iaskari" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/fatihy101"><img src="https://avatars.githubusercontent.com/u/34458068?v=4?s=100" width="100px;" alt="Fatih Yaman"/><br /><sub><b>Fatih Yaman</b></sub></a><br /><a href="#design-fatihy101" title="Design">🎨</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://www.awesomejerry.space"><img src="https://avatars.githubusercontent.com/u/6601073?v=4?s=100" width="100px;" alt="JerryShen"/><br /><sub><b>JerryShen</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=awesomejerry" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/morio77"><img src="https://avatars.githubusercontent.com/u/68191253?v=4?s=100" width="100px;" alt="本多健也"/><br /><sub><b>本多健也</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=morio77" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/TheGlorySaint"><img src="https://avatars.githubusercontent.com/u/21318321?v=4?s=100" width="100px;" alt="Tempelritter"/><br /><sub><b>Tempelritter</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=TheGlorySaint" title="Code">💻</a> <a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=TheGlorySaint" title="Documentation">📖</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/Silfalion"><img src="https://avatars.githubusercontent.com/u/23188369?v=4?s=100" width="100px;" alt="Silfalion"/><br /><sub><b>Silfalion</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=Silfalion" title="Code">💻</a></td>
    </tr>
    <tr>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/guchengxi1994"><img src="https://avatars.githubusercontent.com/u/33513462?v=4?s=100" width="100px;" alt="Chengxi Gu"/><br /><sub><b>Chengxi Gu</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=guchengxi1994" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://www.saifbillah.com"><img src="https://avatars.githubusercontent.com/u/23041420?v=4?s=100" width="100px;" alt="Saif Billah"/><br /><sub><b>Saif Billah</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=saifb" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://github.com/markszente"><img src="https://avatars.githubusercontent.com/u/29143275?v=4?s=100" width="100px;" alt="Mark Szente"/><br /><sub><b>Mark Szente</b></sub></a><br /><a href="https://github.com/subhamayd2/day_night_time_picker/commits?author=markszente" title="Code">💻</a></td>
      <td align="center" valign="top" width="14.28%"><a href="https://linkedin.com/in/rcjuancarlosuwu"><img src="https://avatars.githubusercontent.com/u/67658540?v=4?s=100" width="100px;" alt="Juan Carlos Ramón Condezo"/><br /><sub><b>Juan Carlos Ramón Condezo</b></sub></a><br /><a href="#design-rcjuancarlosuwu" title="Design">🎨</a></td>

CHANGELOG.md
## [1.3.0+1] - 15th August 2023.

- added `settings` prop to pass [RouteSettings] to the [PageRouteBuilder]

## [1.3.0] - 17th June 2023.

- added prop to set the sunrise/sunset/dusk timing
- added `showCancelButton` to display the cancel button or not
- auto focus next selector bug fixed
- added linting

## [1.2.0+2] - 12th March 2023.

- Minor bug fix.

## [1.2.0+1] - 5th March 2023.

- Minor bug fix. **IMPORTANT** for people using the `iosStyle`.

## [1.2.0] - 23nd February 2023.

- _**Includes breaking changes**_
- Refactored code to only have one function i.e. `createPicker`.
  - `createInlinePicker` is now deprecated.
- Added support to `second` input as well
- Renamed prop `disableAutoFocusMinuteAfterHour` to `disableAutoFocusToNextInput` to work with `second` input as well

## [1.1.6] - 22nd February 2023.

- Added prop `width` and `height` for the picker
- Added prop `disableAutoFocusMinuteAfterHour` to disable autofocus to minute after hour is selected

## [1.1.5] - 14th January 2023.

- Fixed where toggling AM/PM was not triggering the onChange

## [1.1.4] - 26th October 2022.

- Added option to hide buttons
- Fixed issue where newly selected time does not update UI

## [1.1.3] - 24th August 2022.

- Added prop `wheelHeight` (only for `createInlinePicker`)
- Fixed typo in README

## [1.1.2] - 23th June 2022.

- Added prop `cancelButtonStyle`
- Added prop `buttonsSpacing`

## [1.1.1] - 14th June 2022.

- Added optional `onCancel` parameter as a callback for the Cancel button

## [1.1.0] - 27th May 2022.

- Added Support for Flutter 3.0
- Added ButtonStyle for `createInlinePicker` and `showPicker`
- Fixed an issue with the ios style picker

## [1.0.5] - 3rd January 2022.

- Fixed overflow issue on smaller devices
- added Bool `ltrMode = true` for ltrMode `false = rtl` on Displaying the TextDirection
- fixed issue where 24HrFormat is not used with iOS Styled Picker

## [1.0.4+1] - 29th December 2021.

- Fixed import
- Refactored

## [1.0.4] - 28th December 2021.

- Separate `TextStyle` for `ok` and `cancel` text.
- Remove adding `.toUpperCase()` to `ok` and `cancel` text.
- Bug fixes
- Refactoring

## [1.0.3+1] - 28th July 2021.

- Added `TextStyle` prop for ok/cancel button.

## [1.0.3] - 28th May 2021.

- Added Thirty in the `MinuteInterval` enum.

## [1.0.2] - 3rd May 2021.

- Workaround fix for `ImageFilter.blur`.

## [1.0.1+1] - 19th April 2021.

- Added prop to auto focus minute picker.

## [1.0.1] - 11th March 2021.

- Added prop to control Dialog padding.

## [1.0.0] - 11th March 2021.

- Null safety!
- Fixed text scale factor.
- Fixed return data on navigator pop()
- Changed FlatButton to TextButton.

## [0.5.0] - 13th February 2021.

- Added option to return value for inline widget on every onValueChange.
- Added option to hide sun/moon animation header.
- Added `themeData` property.
- Minor performance fixes.
- Other bug fixes.

## [0.4.0] - 21st November 2020.

- Added time picker range.
- Added minute interval.
- Enable/disable hour or minute.
- Added option to render as inline widget.
- Other bug fixes.

## [0.3.0+2] - 08th October 2020.

- Update img src in readme. Yeah twice, coz I am stupid.

## [0.3.0+2] - 08th October 2020.

- Update img src in readme. Yeah twice, coz I am stupid.

## [0.3.0+1] - 08th October 2020.

- Update img src in readme

## [0.3.0] - 08th October 2020.

- Added IOS style picker

## [0.2.1+1] - 08th October 2020.

- Displacement issue fix for Sun and Moon assets

## [0.2.1] - 11th August 2020.

- Added optional `unselectedColor` for options

## [0.2.0+1] - 31st July 2020.

- Updated Documentation

## [0.2.0] - 31st July 2020.

- Added optional callback to return data in DateTime.
- Added other bunch of parameters to customize the picker

## [0.1.3+3] - 17th April 2020.

- Barrier color.

## [0.1.3+2] - 16th April 2020.

- Typo.

## [0.1.3+1] - 16th April 2020.

- Minor patches.

## [0.1.3] - 16th April 2020.

- Added more blur customization.

## [0.1.2] - 16th April 2020.

- Minor optimizations.

## [0.1.1] - 3rd April 2020.

- Minor changes related to example project.

## [0.1.0] - 3rd April 2020.

- Initial release.


lib/lib/daynight_timepicker.dart
import 'package:day_night_time_picker/lib/constants.dart';
import 'package:day_night_time_picker/lib/day_night_timepicker_android.dart';
import 'package:day_night_time_picker/lib/day_night_timepicker_ios.dart';
import 'package:day_night_time_picker/lib/state/state_container.dart';
import 'package:day_night_time_picker/lib/state/time.dart';
import 'package:flutter/material.dart';

///
/// The function that shows the *DayNightTimePicker*
///
/// This function takes in the following parameters:
///
/// **value** - `Required` Display value. It takes in [TimeOfDay].
///
/// **onChange** - `Required` Return the new time the user picked as [TimeOfDay].
///
/// **isInlinePicker** - Whether to render an inline widget. Defaults to `false`.
///
/// **onChangeDateTime** - Return the new time the user picked as [DateTime].
///
/// **onCancel** - Custom callback for the Cancel button. Note: if provided, it will override the default behavior of the Cancel button.
///
/// **is24HrFormat** - Show the time in TimePicker in 24 hour format. Defaults to `false`.
///
/// **accentColor** - Accent color of the TimePicker. Defaults to `Theme.of(context).accentColor`.
///
/// **unselectedColor** - Color applied unselected options (am/pm, hour/minute). Defaults to `Colors.grey`.
///
/// **cancelText** - Text displayed for the Cancel button. Defaults to `cancel`.
///
/// **okText** - Text displayed for the Ok button. Defaults to `ok`.
///
/// **sunAsset** - Image asset used for the Sun. Default asset provided.
///
/// **moonAsset** - Image asset used for the Moon. Default asset provided.
///
/// **blurredBackground** - Whether to blur the background of the [Modal]. Defaults to `false`.
///
/// **barrierColor** - Color of the background of the [Modal]. Defaults to `Colors.black45`.
///
/// **borderRadius** - Border radius of the [Container] in `double`. Defaults to `10.0`.
///
/// **elevation** - Elevation of the [Modal] in double. Defaults to `12.0`.
///
/// **dialogInsetPadding** - Inset padding of the [Modal] in EdgeInsets. Defaults to `EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0)`.
///
/// **barrierDismissible** - Whether clicking outside should dismiss the [Modal]. Defaults to `true`.
///
/// **iosStylePicker** - Whether to display a IOS style picker (Not exactly the same). Defaults to `false`.
///
/// **displayHeader** - Whether to display the sun moon animation. Defaults to `true`.
///
/// **hourLabel** - The label to be displayed for `hour` picker. Only for _iosStylePicker_. Defaults to `'hours'`.
///
/// **minuteLabel** - The label to be displayed for `minute` picker. Only for _iosStylePicker_. Defaults to `'minutes'`.
///
/// **secondLabel** - The label to be displayed for `second` picker. Only for _iosStylePicker_. Defaults to `'seconds'`.
///
/// **minuteInterval** - Steps interval while changing `minute`. Accepts `TimePickerInterval` enum. Defaults to `TimePickerInterval.ONE`.
///
/// **secondInterval** - Steps interval while changing `second`. Accepts `TimePickerInterval` enum. Defaults to `TimePickerInterval.ONE`.
///
/// **disableMinute** - Disables the minute picker. Defaults to `false`.
///
/// **disableHour** - Disables the hour picker. Defaults to `false`.
///
/// **maxHour** - Selectable maximum hour. Defaults to `12` | `23`.
///
/// **maxMinute** - Selectable maximum minute. Defaults to `59`.
///
/// **maxSecond** - Selectable maximum second. Defaults to `59`.
///
/// **minHour** - Selectable minimum hour. Defaults to `0` | `1`.
///
/// **minMinute** - Selectable minimum minute. Defaults to `0`.
///
/// **minSecond** - Selectable minimum second. Defaults to `0`.
///
/// **focusMinutePicker** - Whether or not the minute picker is auto focus/selected. Defaults to `false`.
///
/// **themeData** - ThemeData to use for the widget.
///
/// **okStyle** - Ok button's text style. Defaults to `const TextStyle(fontWeight: FontWeight.bold)`.
///
/// **cancelStyle** - Cancel button's text style. Defaults to `const TextStyle(fontWeight: FontWeight.bold)`.
///
/// **hideButtons** - Whether to hide the buttons (ok and cancel). Defaults to `false`.
///
/// **disableAutoFocusToNextInput** - Whether to disable the auto focus to the next input after current input is selected. Defaults to `false`.
///
/// **width** - Fixed width of the Picker container. Defaults to `300` but `350` for `iosStyle`.
///
/// **height** - Fixed height of the Picker container. Defaults to `400`.
///
/// **showSecondSelector** - Whether to use the second selector as well. Defaults to `false`.
///
/// **settings** - Data that might be useful in constructing a [Route].
///
dynamic showPicker({
  Key? key,
  BuildContext? context,
  required Time value,
  required void Function(Time) onChange,
  bool isInlinePicker = false,
  void Function(DateTime)? onChangeDateTime,
  void Function()? onCancel,
  bool is24HrFormat = false,
  Color? accentColor,
  Color? unselectedColor,
  bool isOnChangeValueMode = false,
  String cancelText = 'Cancel',
  String okText = 'Ok',
  Image? sunAsset,
  Image? moonAsset,
  bool blurredBackground = false,
  bool ltrMode = true,
  Color barrierColor = Colors.black45,
  double? borderRadius,
  double? elevation,
  EdgeInsets? dialogInsetPadding =
      const EdgeInsets.symmetric(horizontal: 40.0, vertical: 24.0),
  bool barrierDismissible = true,
  bool iosStylePicker = false,
  bool displayHeader = true,
  String hourLabel = 'hours',
  String minuteLabel = 'minutes',
  String secondLabel = 'seconds',
  TimePickerInterval minuteInterval = TimePickerInterval.ONE,
  TimePickerInterval secondInterval = TimePickerInterval.ONE,
  bool disableMinute = false,
  bool disableHour = false,
  double minMinute = 0,
  double maxMinute = 59,
  double minSecond = 0,
  double maxSecond = 59,
  ThemeData? themeData,
  bool focusMinutePicker = false,
  // Infinity is used so that we can assert whether or not the user actually set a value
  double minHour = double.infinity,
  double maxHour = double.infinity,
  TextStyle okStyle = const TextStyle(fontWeight: FontWeight.bold),
  TextStyle cancelStyle = const TextStyle(fontWeight: FontWeight.bold),
  ButtonStyle? buttonStyle,
  ButtonStyle? cancelButtonStyle,
  double? buttonsSpacing,
  bool hideButtons = false,
  bool disableAutoFocusToNextInput = false,
  double width = 300,
  double height = 400,
  bool showSecondSelector = false,
  double? wheelHeight,
  bool showCancelButton = true,
  sunrise = const TimeOfDay(hour: 6, minute: 0),
  sunset = const TimeOfDay(hour: 18, minute: 0),
  duskSpanInMinutes = 120,
  RouteSettings? settings,
}) {
  if (minHour == double.infinity) {
    minHour = 0;
  }
  if (maxHour == double.infinity) {
    maxHour = 23;
  }

  if (iosStylePicker) {
    width = 350;
  }

  assert(
    !(disableHour == true && disableMinute == true),
    'Both "disableMinute" and "disableHour" cannot be true.',
  );
  assert(
    !(disableMinute == true && focusMinutePicker == true),
    'Both "disableMinute" and "focusMinutePicker" cannot be true.',
  );
  assert(maxMinute <= 59, '"maxMinute" must be less than or equal to 59');
  assert(minMinute >= 0, '"minMinute" must be greater than or equal to 0');
  assert(
    maxHour <= 23 && minHour >= 0,
    '"minHour" and "maxHour" should be between 0-23',
  );

  final timeValue = Time.fromTimeOfDay(value, value.second);

  TimeModelBinding timeModelBinding(child) => TimeModelBinding(
        key: key,
        initialTime: timeValue,
        isInlineWidget: isInlinePicker,
        onChange: onChange,
        onChangeDateTime: onChangeDateTime,
        onCancel: onCancel,
        is24HrFormat: is24HrFormat,
        displayHeader: displayHeader,
        accentColor: accentColor,
        unselectedColor: unselectedColor,
        cancelText: cancelText,
        okText: okText,
        sunAsset: sunAsset,
        moonAsset: moonAsset,
        blurredBackground: blurredBackground,
        borderRadius: borderRadius,
        elevation: elevation,
        dialogInsetPadding: dialogInsetPadding,
        minuteInterval: minuteInterval,
        secondInterval: secondInterval,
        disableMinute: disableMinute,
        disableHour: disableHour,
        maxHour: maxHour,
        maxMinute: maxMinute,
        maxSecond: maxSecond,
        minHour: minHour,
        minMinute: minMinute,
        minSecond: minSecond,
        focusMinutePicker: focusMinutePicker,
        okStyle: okStyle,
        cancelStyle: cancelStyle,
        buttonStyle: buttonStyle,
        cancelButtonStyle: cancelButtonStyle,
        buttonsSpacing: buttonsSpacing,
        hourLabel: hourLabel,
        minuteLabel: minuteLabel,
        secondLabel: secondLabel,
        ltrMode: ltrMode,
        disableAutoFocusToNextInput: disableAutoFocusToNextInput,
        width: width,
        height: height,
        showSecondSelector: showSecondSelector,
        wheelHeight: wheelHeight,
        isOnValueChangeMode: isOnChangeValueMode,
        hideButtons: hideButtons,
        showCancelButton: showCancelButton,
        child: child,
        sunrise: sunrise,
        sunset: sunset,
        duskSpanInMinutes: duskSpanInMinutes,
      );

  if (isInlinePicker) {
    return timeModelBinding(
      Builder(
        builder: (context) {
          if (iosStylePicker) {
            return Builder(
              builder: (context) {
                return Theme(
                  data: themeData ?? Theme.of(context),
                  child: DayNightTimePickerIos(
                    sunrise: sunrise,
                    sunset: sunset,
                    duskSpanInMinutes: duskSpanInMinutes,
                  ),
                );
              },
            );
          } else {
            return Builder(
              builder: (context) {
                return Theme(
                  data: themeData ?? Theme.of(context),
                  child: DayNightTimePickerAndroid(
                    sunrise: sunrise,
                    sunset: sunset,
                    duskSpanInMinutes: duskSpanInMinutes,
                  ),
                );
              },
            );
          }
        },
      ),
    );
  }

  return PageRouteBuilder(
    pageBuilder: (context, _, __) {
      if (iosStylePicker) {
        return Theme(
          data: themeData ?? Theme.of(context),
          child: DayNightTimePickerIos(
            sunrise: sunrise,
            sunset: sunset,
            duskSpanInMinutes: duskSpanInMinutes,
          ),
        );
      } else {
        return Theme(
          data: themeData ?? Theme.of(context),
          child: DayNightTimePickerAndroid(
            sunrise: sunrise,
            sunset: sunset,
            duskSpanInMinutes: duskSpanInMinutes,
          ),
        );
      }
    },
    transitionDuration: const Duration(milliseconds: 200),
    transitionsBuilder: (context, anim, secondAnim, child) => SlideTransition(
      position: anim.drive(
        Tween(
          begin: const Offset(0, 0.15),
          end: const Offset(0, 0),
        ).chain(
          CurveTween(curve: Curves.ease),
        ),
      ),
      child: FadeTransition(
        opacity: anim,
        child: timeModelBinding(child),
      ),
    ),
    barrierDismissible: barrierDismissible,
    opaque: false,
    barrierColor: barrierColor,
    settings: settings,
  );
}

@Deprecated(
  'Please pass boolean prop `isInlinePicker` to `showPicker` to render an inline picker',
)
void createInlinePicker() {}
